OpenRA.Game/Renderer.cs

```diff
@@ -267,6 +269,8 @@ public void BeginWorld(Rectangle worldViewport)
 				lastWorldViewport = worldViewport;
 			}
 
+			this.visualOffset = visualOffset;
```

Comment:
IMO its probably cleaner to use `worldSprite.Offset` for this instead of tracking it outside the sprite (which does mean creating a new sprite when the offset changes, but sprites are frequently created and thrown away everywhere else in the engine so caching this particular one probably isn't needed).

Reply:
refactored